plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/rename.kt

```diff
@@ -28,55 +31,90 @@ class RenameInto : AbstractSchemaModificationInterpreter() {
-    return PluginDataFrameSchema(
-        f(columns(), nextName, selected, emptyList())
+internal fun PluginDataFrameSchema.map(selected: ColumnsSet, nextName: () -> String): PluginDataFrameSchema =
+    PluginDataFrameSchema(
+        f(columns(), nextName, selected, emptyList()),
     )
-}
 
-internal fun f(columns: List<SimpleCol>, nextName: () -> String, selected: ColumnsSet, path: List<String>): List<SimpleCol> {
-    return columns.map {
+internal fun f(
+    columns: List<SimpleCol>,
+    nextName: () -> String,
+    selected: ColumnsSet,
+    path: List<String>,
+): List<SimpleCol> =
+    columns.map {
         val fullPath = path + listOf(it.name)
         when (it) {
             is SimpleColumnGroup -> {
                 val group = if (fullPath in selected) {
                     it.rename(nextName())
-                }  else {
+                } else {
                     it
                 }
                 group.map(selected, fullPath, nextName)
             }
+
             is SimpleFrameColumn -> {
                 val frame = if (fullPath in selected) {
                     it.rename(nextName())
-                }  else {
+                } else {
                     it
                 }
                 frame.map(selected, fullPath, nextName)
             }
+
             is SimpleDataColumn -> if (fullPath in selected) {
                 it.rename(nextName())
             } else {
                 it
             }
         }
     }
-}
 
-internal fun SimpleColumnGroup.map(selected: ColumnsSet, path: List<String>, nextName: () -> String): SimpleColumnGroup {
-    return SimpleColumnGroup(
-        name,
-        f(columns(), nextName, selected, path)
+internal fun SimpleColumnGroup.map(
+    selected: ColumnsSet,
+    path: List<String>,
+    nextName: () -> String,
+): SimpleColumnGroup =
+    SimpleColumnGroup(
+        name = name,
+        columns = f(columns(), nextName, selected, path),
     )
-}
 
-internal fun SimpleFrameColumn.map(selected: ColumnsSet, path: List<String>, nextName: () -> String): SimpleFrameColumn {
-    return SimpleFrameColumn(
-        name,
-        f(columns(), nextName, selected, path)
+internal fun SimpleFrameColumn.map(
+    selected: ColumnsSet,
+    path: List<String>,
+    nextName: () -> String,
+): SimpleFrameColumn =
+    SimpleFrameColumn(
+        name = name,
+        columns = f(columns(), nextName, selected, path),
     )
+
+class RenameToCamelCase : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by dataFrame()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema =
+        receiver.asDataFrame().renameToCamelCase().toPluginDataFrameSchema()
+}
+
+class RenameToCamelCaseClause : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver: RenameClauseApproximation by arg()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
```

Comment:
i'd suggest convert selected columns to path and `receiver.df.asDataFrame().rename { paths }.toCamelCase().toPluginDataFrameSchema()`

Reply:
and it promotes the idea that the original function solves the compiler plugin part :) which is great. I just need to become a bit more comfortable with the helper functions available to me in the compiler plugin module, but I'll get there :)